src/helpers/filling.rs

```diff
@@ -97,3 +100,78 @@ pub fn fill_tilemap_rect_color(
         }
     }
 }
+
+/// Generates a vector of hex positions that form a ring of given `radius` around the specified
+/// `origin`.
+///
+/// If `radius` is zero, `origin` is the only position in the returned vector.
+pub fn generate_hex_ring(origin: AxialPos, radius: u32) -> Vec<AxialPos> {
+    if radius == 0 {
+        vec![origin]
+    } else {
+        let mut ring = Vec::with_capacity((radius * 6) as usize);
+        let corners = HEX_DIRECTIONS
+            .iter()
+            .map(|direction| origin + radius * AxialPos::from(direction))
+            .collect::<Vec<AxialPos>>();
+        // The "tangent" is the direction we must travel in to reach the next corner
+        let tangents = (0..6)
+            .map(|ix| HexDirection::from(ix + 2).into())
+            .collect::<Vec<AxialPos>>();
+
+        for (&corner, &tangent) in corners.iter().zip(tangents.iter()) {
+            for k in 0..radius {
+                ring.push(corner + k * tangent);
+            }
+        }
+
+        ring
+    }
+}
+
+/// Generates a vector of hex positions that form a hexagon of given `radius` around the specified
+/// `origin`.
+pub fn generate_hexagon(origin: AxialPos, radius: u32) -> Vec<AxialPos> {
+    let mut hexagon = Vec::with_capacity((6 * radius * (radius + 1) / 2) as usize);
+    for r in 0..radius {
+        hexagon.extend(generate_hex_ring(origin, r));
+    }
+    hexagon
+}
+
+/// Fills a hexagonal region with the given `tile_texture`.
+///
+/// The rectangular region is defined by an `origin` in [`TilePos`](crate::tiles::TilePos), and a
+/// `radius`.
+///
+/// Tiles that do not fit in the tilemap will not be created.
+pub fn fill_tilemap_hexagon(
```

Comment:
It might be worth mentioning the cost here? This seems like O(n^2) to me.

Reply:
Yeah, this is pretty inescapably (and obviously) n^2. I don't think it should be called out.

Reply:
> Yeah, this is pretty inescapably (and obviously) n^2. I don't think it should be called out.
agreed